### PR TITLE
Stochastic Gradient Descent Optimizer

### PR DESCRIPTION
**Pull request by Transurgeon**
WIP branch for issue #6

### CODE REVIEW

File: yadll/tensor/optimizers/optimizer.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    def step(self):
+        pass
+
+    def zero_grad(self):
+        for p in self.params:
+            p.grad = None
+
+
+class SGD(Optimizer):
+    def __init__(self, params, lr: float, momentum: float):
```
Comment (Transurgeon) [inline yadll/tensor/optimizers/optimizer.py:19]: I added momentum here so that the args count is the same, but it isn't actually doing anything.
We need to figure out why the ``test_sgd_momentum`` unit-test is passing

File: test/test_tensor/test_tensor_autodiff.py [hunk 2]
```diff
@@ -2,46 +2,49 @@
 import numpy as np
 import torch
 
+
```
Comment (Transurgeon) [inline test/test_tensor/test_tensor_autodiff.py:5]: so these big changes are all due to the reformatting I did using pycharm in my local. I am not sure it is relevant, and I think with our addition of black as a CI linter, these updates won't be needed anymore. Should I look into removing these changes?

File: test/test_optimizers/test_sgd.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+
+def test_sgd_with_momentum():
+    x = Tensor.random((2, 2))
+    torch_x = torch.tensor(x.data)
+    model = Linear(2, 1, False)
+    torch_model = torch.nn.Linear(2, 1, False)
+    torch_model.weight = torch.nn.Parameter(torch.tensor(model.weight.data))
+    optim = SGD(model.parameters(), 0.1, 0.9)
+    torch_optim = torch.optim.SGD(torch_model.parameters(), 0.1, 0.9)
```
Comment (Transurgeon) [inline test/test_optimizers/test_sgd.py:34]: Given that we are changing only the momentum value between the two tests, we could potentially use [pytest.parametrize](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html) to remove some lines. 
I understand that we are testing two different things, but it could suffice to simply mention as a comment that when the last arg = 0, it means there is no momentum.

File: test/test_tensor/test_tensor_autodiff.py (hunk 2 above)
Comment (fredpell1) [reply test/test_tensor/test_tensor_autodiff.py:5]: it's ok, we will squash the commits anyway

File: test/test_optimizers/test_sgd.py (hunk 3 above)
Comment (fredpell1) [reply test/test_optimizers/test_sgd.py:34]: for some tests yeah, but with the recent push they're too different

**Comment by fredpell1** [comment]
merging, tests are passing locally
